### PR TITLE
001 view job roles

### PR DESCRIPTION
**Pull request by PauricDonnelly**

### CODE REVIEW

**Comment by laura-puckoriute** [comment]
Hi Pauric, please rename the attributes (and update getters and setters) in the **`JobRoles`** java class so that it matches the database fields in Role table.

- [x] jobTitle -> title

- [x] band -> band_level

and `@JsonProperty` in the same file:

- [x] jobtitle -> title

- [x] band -> band_level

And accordingly update the Jobs.java file.

**Review by laura-puckoriute — APPROVED** [review]

**Review by laura-puckoriute — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -35,4 +35,5 @@ build/
 .vscode/
 
 ### Mac OS ###
-.DS_Store
\ No newline at end of file
+.DS_Storedb.properties
```
Comment (shaunganley) [inline .gitignore]: this line looks wrong

File: src/main/java/org/kainos/ea/controller/Jobs.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.kainos.ea.models.JobRoles;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Jobs {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/controller/Jobs.java]: this file should be in a data folder, not a controller

File: src/test/java/org/kainos/ea/service/JobsTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+        Mockito.when(jobs.getJobRoles(conn)).thenReturn(expectedResult);
+        List<JobRoles> result = jobsService.getJobRoles();
+
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    void getJobRoles_shouldThrowSQLException_whenJobsThrowsSQLException () throws SQLException, DatabaseConnectionException {
```
Comment (shaunganley) [inline src/test/java/org/kainos/ea/service/JobsTest.java]: you should also test for DatabaseConnectionException

File: src/main/java/org/kainos/ea/controller/Jobs.java (hunk 2 above)
Comment (shaunganley) [reply src/main/java/org/kainos/ea/controller/Jobs.java]: also can you rename this to some like JobRolesData

**Review by shaunganley — CHANGES_REQUESTED** [review]
small changes requested

**Review by shaunganley — APPROVED** [review]

**Pull request merged**